server/src/test/scala/org/http4s/server/middleware/DefaultHeadSpec.scala

```diff
@@ -3,13 +3,17 @@ package server
 package middleware
 
 import cats.effect._
-import cats.effect.concurrent.Ref
+import cats.effect.concurrent.Deferred
 import fs2.Stream
 import org.http4s.Uri.uri
 import org.http4s.dsl.io._
 import org.http4s.testing.Http4sLegacyMatchersIO
+import cats.effect.testing.specs2.CatsIO
+
+class DefaultHeadSpec extends Http4sSpec with Http4sLegacyMatchersIO with CatsIO {
```

Comment:
If you extend `CatsEffect` instead of `CatsIO`, you don't need the two overrides below.